src/storage/gcloud-storage/gcp-storage.service.ts

```diff
@@ -119,6 +121,41 @@ export class GCPStorageService {
         return ok(true);
     }
 
+    private async compressFile(file: File): Promise<Buffer> {
+        const chunks: Buffer[] = [];
+
+        for await (const chunk of file.stream()) {
+            chunks.push(Buffer.from(chunk));
+        }
+        const fileBuffer = Buffer.concat(chunks);
+
+        try {
+            const sharpPipeline = sharp(fileBuffer).resize({
+                width: 1200,
+                withoutEnlargement: true,
+            });
+
+            const format = file.type.split('/')[1];
+
+            if (format === 'jpeg' || format === 'jpg') {
+                return sharpPipeline.jpeg({ quality: 75 }).toBuffer();
+            }
+
+            if (format === 'png') {
+                return sharpPipeline.png({ compressionLevel: 9 }).toBuffer();
+            }
+
+            if (format === 'webp') {
+                return sharpPipeline.webp({ quality: 75 }).toBuffer();
+            }
+
+            return fileBuffer;
+        } catch (error) {
+            // Return buffer on the original file if compression failed
+            return fileBuffer;
```

Comment:
Do we want to log an error here or something?

Reply:
Good one, fixed in 96b89d10fb4e3b9362da3081a985d3c86903a102